fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java

```diff
@@ -83,48 +84,51 @@ public FilterEstimation(Set<Slot> aggSlots) {
-            Statistics orStats = context.statistics.withRowCount(rowCount);
-            for (Map.Entry<Expression, ColumnStatistic> entry : orStats.columnStatistics().entrySet()) {
-                ColumnStatistic leftColStats = leftStats.findColumnStatistics(entry.getKey());
-                ColumnStatistic rightColStats = rightStats.findColumnStatistics(entry.getKey());
-                ColumnStatisticBuilder estimatedColStatsBuilder = new ColumnStatisticBuilder(entry.getValue());
-                if (leftColStats.minValue <= rightColStats.minValue) {
-                    estimatedColStatsBuilder.setMinValue(leftColStats.minValue);
-                    estimatedColStatsBuilder.setMinExpr(leftColStats.minExpr);
-                } else {
-                    estimatedColStatsBuilder.setMinValue(rightColStats.minValue);
-                    estimatedColStatsBuilder.setMinExpr(rightColStats.minExpr);
-                }
-                if (leftColStats.maxValue >= rightColStats.maxValue) {
-                    estimatedColStatsBuilder.setMaxValue(leftColStats.maxValue);
-                    estimatedColStatsBuilder.setMaxExpr(leftColStats.maxExpr);
-                } else {
-                    estimatedColStatsBuilder.setMaxValue(rightColStats.maxValue);
-                    estimatedColStatsBuilder.setMaxExpr(rightColStats.maxExpr);
+            Statistics orStats = context.statistics.setRowCount(rowCount);
+            Set<Slot> leftInputSlots = leftExpr.getInputSlots();
+            Set<Slot> rightInputSlots = rightExpr.getInputSlots();
+            for (Slot slot : context.keyColumns) {
+                if (leftInputSlots.contains(slot) && rightInputSlots.contains(slot)) {
+                    ColumnStatistic leftColStats = leftStats.findColumnStatistics(slot);
+                    ColumnStatistic rightColStats = rightStats.findColumnStatistics(slot);
+                    StatisticRange leftRange = StatisticRange.from(leftColStats, slot.getDataType());
+                    StatisticRange rightRange = StatisticRange.from(rightColStats, slot.getDataType());
+                    StatisticRange union = leftRange.union(rightRange);
+                    ColumnStatisticBuilder colBuilder = new ColumnStatisticBuilder(
+                            context.statistics.findColumnStatistics(slot));
+                    colBuilder.setMinValue(union.getLow()).setMinExpr(union.getLowExpr())
+                            .setMaxValue(union.getHigh()).setMaxExpr(union.getHighExpr())
+                            .setNdv(union.getDistinctValues());
+                    orStats.addColumnStats(slot, colBuilder.build());
                 }
             }
             return orStats;
         }
+        // should not come here
+        Preconditions.checkArgument(false, "unsupported compound operator: "
+                + predicate.getClass().getName() + " in " + predicate.toSql());
```

Comment:
Maybe throw a runtime exception is better

Reply:
Preconditions.checkArgument() throws runtime exception.